plugins/woocommerce/src/Logging/RemoteLogger.php

```diff
@@ -16,10 +16,202 @@
  * @since 9.2.0
  * @package WooCommerce\Classes
  */
-class RemoteLogger {
+class RemoteLogger extends \WC_Log_Handler {
+	const LOG_ENDPOINT                = 'https://public-api.wordpress.com/rest/v1.1/logstash';
+	const THROTTLE_TRANSIENT          = 'woocommerce_remote_logging_throttle';
+	const THROTTLE_LIMIT              = 10; // Maximum number of requests in the interval.
+	const THROTTLE_INTERVAL           = 300; // Throttle interval in seconds (5 minutes).
 	const WC_LATEST_VERSION_TRANSIENT = 'latest_woocommerce_version';
 	const FETCH_LATEST_VERSION_RETRY  = 'fetch_latest_woocommerce_version_retry';
 
+	/**
+	 * The logger instance.
+	 *
+	 * @var \WC_Logger_Interface|null
+	 */
+	private $local_logger;
+
+	/**
+	 * Remote logger constructor.
+	 *
+	 * @internal
+	 * @param \WC_Logger_Interface|null $logger Logger instance.
+	 */
+	public function __construct( \WC_Logger_Interface $logger = null ) {
+		if ( null === $logger ) {
+			$this->local_logger = wc_get_logger();
+		}
+		$this->local_logger = $logger;
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param int    $timestamp Log timestamp.
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug.
+	 * @param string $message Log message.
+	 * @param array  $context Optional. Additional information for log handlers.
+	 *
+	 * @return bool False if value was not handled and true if value was handled.
+	 */
+	public function handle( $timestamp, $level, $message, $context ) {
```

Comment:
Is there a reason why we want to extend WC_Log_Handler when we're not implementing the interface method `handle`?  

It seems like classes that extends `WC_Log_Handler` are used in https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-logger.php#L167

I think RemoteLogger is somewhat specialized logger that is called directly in https://github.com/woocommerce/woocommerce/pull/49599/files#diff-97fb927371ee663e98afaf8b8134905a96d4b59c68f6a83c569ec42a6b468a08R405 and might not need to extend/implement [WC_Log_Handler](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/abstracts/abstract-wc-log-handler.php) or [WC_Log_Handler_Interface](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/interfaces/class-wc-log-handler-interface.php)

Reply:
Good question! I extended `WC_Log_Handler` because I wanted to reuse `get_backtrace` method from WC_Log_Handler. I could have just copied the method to RemoteLogger but I thought it would be better to reuse it. What do you think?

https://github.com/woocommerce/woocommerce/blob/12b701d35cac2c7b560d1685f485ce8283f4dc85/plugins/woocommerce/includes/abstracts/abstract-wc-log-handler.php#L66-L97

Reply:
Ah, I see! I don't think there is any harm in extending it, as we'll most likely use RemoteLogger in a specific place.

My comment was purely from a technical point of view.

LGTM!